addons/web/static/src/views/list/list_renderer.js

```diff
@@ -1059,6 +1073,34 @@ export class ListRenderer extends Component {
         return null;
     }
 
+    expandCheckboxes(record, direction) {
+        const recordIndex = this.props.list.records.indexOf(record);
```

Comment:
This comment applies to the whole diff you made in this file.

nitpicking: IMO when you access same value more than once in nested objects, it is more readable to destructure. Furthermore, It is a micro optimization.
```suggestion
const { records } = this.props.list;
const recordIndex = records.indexOf(record);
```
then replace `this.props.list.records` below with just `records`